### ISSUE TITLE
Build errors building iOS apps with `bazel coverage`

### ISSUE THREAD

**Issue by jerrymarino** [bazel-ios/rules_ios#491]
Bazel is now able to pass the right codecov flags to clang invocations when running, however it's unclear what the status is to get this working end to end.

While doesn't seem strictly related to rules, I get the following errors when building with coverage:
```
bazel coverage -s tools/TestBit ...

Undefined symbols for architecture arm64:
  "_llvm_gcov_init", referenced from:
      ___llvm_gcov_init in libTestBit.a(TestBit.o)
  "_llvm_gcda_end_file", referenced from:
      ___llvm_gcov_writeout in libTestBit.a(TestBit.o)
  "_llvm_gcda_emit_function", referenced from:
      ___llvm_gcov_writeout in libTestBit.a(TestBit.o)
  "_llvm_gcda_emit_arcs", referenced from:
      ___llvm_gcov_writeout in libTestBit.a(TestBit.o)
  "_llvm_gcda_start_file", referenced from:
      ___llvm_gcov_writeout in libTestBit.a(TestBit.o)
  "_llvm_gcda_summary_info", referenced from:
      ___llvm_gcov_writeout in libTestBit.a(TestBit.o)
ld: symbol(s) not found for architecture arm64
```

Upon digging into this, it looks like I can pass `-fprofile-instr-generate`. To get this building, I can run Bazel on the CLI as such:

```
bazel coverage -s tools/TestBit --linkopt=-fprofile-instr-generate
```
However, we should make sure that this is fully working end to end in some capacity, as there are other issues. On another build, I'm getting:
```
# Execution platform: @local_config_platform//:host
error: failed to open coverage notes file for writing: Permission denied
```
Because it is trying to overwrite an existing `.gcno` inside of bazel-out! Removing that allows it to re-build.

**Comment by keith** [comment]
probably missing `--experimental_use_llvm_covmap`

**Comment by jerrymarino** [comment]
@keith  ok awesome, I'll give it a try! It looks like a [_ton_](https://bazel.build/docs/coverage) of improvements had went into Bazel for codecov in general in the last year. That sees to clear up the linker issue, I've still get another error, which makes me think there is some configuration but that causes the aspect to fail:
```
        File "/private/var/tmp/_bazel_jmarino/ca0596bbcafdd3d97374bd4573e006cf/external/build_bazel_rules_apple/apple/internal/testing/apple_test_rule_support.bzl", line 86, column 38, in _coverage_files_aspect_impl
                covered_binaries = depset(
Error in depset: cannot add an item of type 'NoneType' to a depset of 'File'
```

Do you know of any posts about folks codecov setup with Bazel?

**Comment by keith** [comment]
Is this with a custom rule? I haven't seen this one before. And I would expect the tests to always be able to find _some_ binary https://github.com/bazelbuild/rules_apple/blob/ac43c1e467564d9df6b3355ff93fcaf224f2c0f9/apple/internal/testing/apple_test_rule_support.bzl#L63-L69

I think most of the changes to this kinda stuff have just been implementation details, so I don't think there's any more info out there

**Comment by jerrymarino** [comment]
Yeah it looks like @mattrobmattrob fixed that issue on a later version, this atleast gets it building. I'll have to catch up on the recent changes to test coverage in general - all the coverage files have empty output, going to circle back with the latest stuff and update https://github.com/bazelbuild/rules_apple/issues/338
